packages/x-date-pickers/src/MonthCalendar/PickersMonth.tsx

```diff
@@ -64,7 +68,7 @@ const PickersMonthButton = styled('button', {
     { [`&.${pickersMonthClasses.selected}`]: styles.selected },
   ],
 })<{
-  ownerState: PickersMonthProps;
+  ownerState?: PickersMonthProps;
```

Comment:
Can it ever be undefined?
It seems that we are passing it on L176...

Reply:
The problem comes from `useSlotProps` which returns a nullable `ownerState` based on some condition on the component:

```ts
    ownerState: ElementType extends React.ComponentType<any> ? OwnerState : ElementType extends keyof JSX.IntrinsicElements ? undefined : OwnerState | undefined;
```

No idea why :laughing: 

```ts
  // With thse the ownerState can be undefined
  monthButton?: React.ElementType<React.HTMLAttributes<HTMLButtonElement>>;
  monthButton?: React.ElementType<React.HTMLAttributes<HTMLButtonElement>, 'button'>;

  // With this the ownerState can't be undefined
  monthButton?: React.ElementType;
```

I can override the type returned by `useSlotProps` though..

Reply:
Well, I'm just thinking about the public API:
Is it available in `styleOverrides`? If so, then it would be great to have a correct type—non-nullable `ownerState`.

Reply:
> Is it available in styleOverrides

For me TS there is no bridge for TS to understand from where each style override comes and to type it.
For no it should no impact it :+1: 

I reduced the typing precision of the slot to remove the error